pkg/controller/jenkins/configuration/user/seedjobs/seedjobs.go

```diff
@@ -160,7 +162,21 @@ func (s *SeedJobs) EnsureSeedJobs(jenkins *v1alpha2.Jenkins) (done bool, err err
 	}
 
 	if len(jenkins.Spec.SeedJobs) > 0 {
-		err := s.createAgent(s.jenkinsClient, s.Client, jenkins, jenkins.Namespace, AgentName)
+		requeue, err := s.ensureAgentExecutorsAmount()
+		if err != nil {
+			return false, err
+		}
+		if requeue {
+			return false, nil
+		}
+
+		// TODO (jakalkhalili): Check the difference in amount of executors. Maybe annotations on agent deployment?
+		err = s.redeployAgent(s.jenkinsClient, s.Client, jenkins, jenkins.Namespace, AgentName)
+		if err != nil {
+			return false, err
+		}
```

Comment:
remove and add annotation with the amount of executors, than the createAgent will update the deployment and new pod will be created only if there was the change of number of executors